src/interactionManager/interactionManager.ts

```diff
@@ -45,9 +45,7 @@ export class InteractionManager{
     return interaction
   }
 
-  // FIXME this can return UNDEFINED, should throw an error
-  // FIXME need to be async to support storage backends
-  public getInteraction<F extends Flow<any> = Flow<any>>(id: string) {
-    return this.interactions[id] as Interaction<F>
+  public async getInteraction<F extends Flow<any>>(id: string, transportAPI?: TransportAPI): Promise<Interaction<F>> {
+    return await Interaction.fromMessages<F>(await this.ctx.storage.get.interactionTokens({ nonce: id }), this, id, transportAPI)
```

Comment:
Should we rather rebuild the interaction from storage only in case we didn't find it in the `this.interactions` map? For instance, in SW1.x/2.0 we use `getInteraction` extensively due to passing around the interaction id. Would be great if we could keep `getInteraction` simple (and synchronous).

Reply:
we could keep checking in `this.interactions` first, but the function would still have to be `async` to allow for checking the storage as well if not found. alternatively, they could be 2 separate methods, but every time you would be calling one (get from `this.interactions` - sync) and then the other (get from storage - async) anyway, so your call site would still have to be `async`. IMO there should be no `.interactions` member on `InteractionManager` at all